doc/builders/special/runinmkshell.section.md

```diff
@@ -0,0 +1,64 @@
+# pkgs.runInMkShell {#sec-pkgs-runInMkShell}
+
+`pkgs.runInMkShell` is a function that generates a wrapper script that
+receives the command to be run as arguments and then run it inside a special
+environment created with [`mkShell`](#sec-pkgs-mkShell).
+
+It's basically like the `--run` flag of `nix-shell`, but using Nix expressions.
```

Comment:
I would like this to be the other way around. Currently Nix has suboptimal, yet ossified logic to implement "shells" because it couples with Nixpkgs stdenv. By flipping this around and making this function the default nix shell implementation, we could solve that architectural problem and let the behavior of `nix-shell`/`nix develop` evolve in a future proof way by pinning/locking Nixpkgs.
This would happen through an indirection such as a `pkg.devShell` attribute, e.g.
- #206728

This idea was received quite positively by the Nix team (even though we don't have an approved issue for it - I think we discussed it before the label).

It's fine to describe it this way until `nix` uses the `devShell` (or similarly named) attribute, so no change required here.